lib/broccoli/default-packager.js

```diff
@@ -844,34 +845,46 @@ module.exports = class DefaultPackager {
         annotation: 'Tests To Process',
       });
 
+
       if (this.isModuleUnificationEnabled) {
 
         let destDir,
             testSrcTree,
-            srcDir;
+            srcDir,
+            dirNotFound;
 
         // ember-addon
         if (this.name === 'dummy') {
           testSrcTree = 'src';
           destDir = `${this.project.name()}/src`;
+          // Classic Addon with MU dummy app `kitchen-sink-with-mu-dummy-app` will throw: `Dir not found: src` exception
+          // `dirNotFound` will skip this error. Find better way to avoid this error.
+
+          // also, in tests we pass root path to this function
+          // in default-packager/tests-test.js
+          const srcPath = typeof tree === 'string' ? `${tree}/src` : 'src';
```

Comment:
@rwjblue This is the same issue we discussed at https://github.com/ember-cli/ember-cli/pull/8322#discussion_r255207949


**The error is triggered by `broccoli/lib/builder` before `BroccoliFunnel.build` is executed; the `BroccoliFunnel.allowEmpty` routine is never executed.**


I can investigate this but I need guidance on where to start.

Reply:
@patricklx This should be fixed by https://gist.github.com/ppcano/78e14886c7a660d17eadab709802adb0

Would you mind to include it and check if all the tests pass.